aten/src/ATen/native/SoftMax.cpp

```diff
@@ -233,29 +233,65 @@ Tensor log_softmax_backward_cpu(
 }
 
 Tensor softmax(const Tensor& input_, const int64_t dim_) {
-  return at::_softmax(input_, dim_, false);
+  auto result = [&]() {
+#ifdef BUILD_NAMEDTENSOR
+    NoNamesGuard guard;
+#endif
+    return at::_softmax(input_, dim_, false);
+  }();
+#ifdef BUILD_NAMEDTENSOR
+  namedinference::propagate_names(result, input_);
+#endif
+  return result;
 }
 
 Tensor softmax(const Tensor& input_, const int64_t dim_, c10::optional<ScalarType> dtype) {
-  if (input_.is_cuda() && input_.scalar_type() == ScalarType::Half && dtype == ScalarType::Float){
-      return at::_softmax(input_, dim_, true);
-  } else {
-      Tensor converted = dtype.has_value() ? input_.toType(dtype.value()) : input_;
-      return at::_softmax(converted, dim_, false);
-  }
+  auto result = [&]() {
+#ifdef BUILD_NAMEDTENSOR
+    NoNamesGuard guard;
+#endif
+    if (input_.is_cuda() && input_.scalar_type() == ScalarType::Half && dtype == ScalarType::Float){
+        return at::_softmax(input_, dim_, true);
+    } else {
+        Tensor converted = dtype.has_value() ? input_.toType(dtype.value()) : input_;
+        return at::_softmax(converted, dim_, false);
+    }
+  }();
+#ifdef BUILD_NAMEDTENSOR
+  namedinference::propagate_names(result, input_);
```

Comment:
is this adding named tensor support that didn't previously exist? from the description it sounds like the intent is just to refactor but I don't know where this is getting moved from. 

Does `at::_softmax` have propagated names without the name guard? if so should we remove it since we're name guarding it anyway?

Reply:
`at::_softmax` propagates names "coincidentally" because it calls `empty_like` under the hood. Named tensor support for softmax existed before this PR but still exists after this PR.

We need to explicitly propagate names now because the returned tensor from the NoNamesGuard block has no names.

Reply:
ah, I was expecting to see a propagate_names removed from `_softmax` but that makes sense